src/Parser.jl

```diff
@@ -272,7 +273,7 @@ function parse_string(ps::ParserState)
         if c == BACKSLASH
             c = advance!(ps)
             if c == LATIN_U  # Unicode escape
-                append!(b, Vector{UInt8}(string(read_unicode_escape!(ps))))
+                append!(b, get_bytes(string(read_unicode_escape!(ps))))
```

Comment:
`append!(b, codeunits(s))` would work here … no need to force a copy to be made as in `get_bytes`

Reply:
The problem was that `codeunits` wasn't defined in `Compat.jl` to make it work on v0.6.

Reply:
Even using `codeunits` is not the best approach here.
Instead of having to create a string just to append it, it would be better to pass `b` to a `read_unicode_escape!(b, ps)` function, that directly `push!`ed the 1-4 bytes from the unicode escape.

Reply:
Another question - why do you think my change would force a copy?  It uses `Vector{UInt8}` on `v0.6`, which doesn't copy (and hence could be unsafe), and `codeunits` on `v0.7`, which doesn't copy either.
I think your objection was incorrect.